core/src/main/scala/org/apache/spark/deploy/history/ApplicationHistoryProvider.scala

```diff
@@ -73,4 +73,11 @@ private[history] abstract class ApplicationHistoryProvider {
   @throws(classOf[SparkException])
   def writeEventLogs(appId: String, attemptId: Option[String], zipStream: ZipOutputStream): Unit
 
+  /**
+   * Read in the event logs from zip input stream provided and replayed by history provider.
+   * @param zipStream Input zip stream.
+   * @throws SparkException if there's any exception read in the zip stream.
+   */
+  @throws(classOf[SparkException])
+  def readEventLogs(zipStream: ZipInputStream): Unit
```

Comment:
Rather than make this abstract, can you add a stub method which just goes `throw new SparkException("Unsupported Feature")`. That way any existing history provider will still load